invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -409,7 +409,6 @@ def deposit_create(community=None):
     )
 
 
-@login_required
```

Comment:
since we remove this, I think we should consider also this scenario:
1. User was logged in, editing the draft
2. User's session expired
3. User refreshed the page
4. Got permission denied, and the fact of expired session is not very visible in the interface, so it can be interpreted as "I had access before and now I don't, what is going on?"

shouldn't we instead check the token in the link if it allows for edits? And then:
if no token-> redirect to login 
token with insufficient permissions -> raise PermissionDenied()
login with insufficient permissions -> raise PermissionDenied()
login with OK permissions or token with OK permissions -> show the view

Reply:
We briefly discussed, we should indeed have an enhanced decorator as suggested by @kpsherva that will simply do:
```
def token_or_login_required():
  if not token:
    login_required
```